### PR TITLE
build-sys: introduce pedantic revive config

### PR DESCRIPTION
**Pull request by karasz**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -8,8 +8,12 @@ GOGENERATE_FLAGS = -v
 GOPATH ?= $(shell $(GO) env GOPATH)
 GOBIN ?= $(GOPATH)/bin
 
+PEDANTIC = 0
```
Comment (amery) [inline Makefile:11]: shouldn't this be `PEDANTIC ?= 0` ?

File: Makefile [hunk 2]
```diff
@@ -8,8 +8,12 @@ GOGENERATE_FLAGS = -v
...
+ifeq ($(PEDANTIC),1)
+    PEDANTIC_ARG ?= -config $(shell pwd)/pedantic.toml
```
Comment (amery) [inline Makefile:13]: `tools/revive-pedantic.toml` ?

**Review by amery — CHANGES_REQUESTED** [review]

File: Makefile (hunk 1 above)
Comment (karasz) [reply Makefile]: yes it should

File: Makefile (hunk 2 above)
Comment (karasz) [reply Makefile]: yup

**Comment by amery** [comment]
we need to fix _master_ before merging this.
```
⚠ 103 problems (0 errors, 103 warnings)

Warnings:
  30  unhandled-error       
  20  comment-spacings      
  11  empty-lines           
  7   unused-parameter      
  6   import-shadowing      
  5   cognitive-complexity  
  4   use-any               
  3   line-length-limit     
  3   unexported-naming     
  3   nested-structs        
  3   empty-block           
  2   flag-parameter        
  2   confusing-naming      
  1   cyclomatic            
  1   unused-receiver       
  1   confusing-results     
  1   function-length   
  ```

File: Makefile [hunk 3]
```diff
@@ -8,8 +8,13 @@ GOGENERATE_FLAGS = -v
...
 GOBIN ?= $(GOPATH)/bin
 
+PEDANTIC ?= 0
+PEDANTIC_CONF ?= -config $(CURDIR)/tools/revive-default.toml
+ifeq ($(PEDANTIC),1)
+    PEDANTIC_CONF = -config $(CURDIR)/tools/revive-pedantic.toml
+endif
 REVIVE ?= $(GOBIN)/revive
-REVIVE_RUN_ARGS ?= -formatter friendly -set_exit_status
+REVIVE_RUN_ARGS ?= $(PEDANTIC_CONF) -formatter friendly
```
Comment (amery) [inline Makefile:17]: `-config $(REVIVE_CONF)` ?

File: tools/revive-default.toml [hunk 4]
```diff
@@ -0,0 +1,30 @@
+ignoreGeneratedHeader = false
+severity = "warning"
+confidence = 0.8
+errorCode = 0
```
Comment (amery) [inline tools/revive-default.toml:4]: shouldn't errors be fatal even on non-pedantic?

File: Makefile (hunk 3 above)
Comment (karasz) [reply Makefile]: resolved

**Review by amery — APPROVED** [review]

**Pull request merged**